Classes/common/query/CDTQQuerySqlTranslator.m

```diff
@@ -495,4 +545,32 @@ + (CDTQSqlParts *)selectStatementForAndClause:(NSArray *)clause usingIndex:(NSSt
     return parts;
 }
 
++ (CDTQSqlParts *)selectStatementForTextClause:(NSObject *)clause usingIndex:(NSString *)indexName
+{
+    if (!clause) {
+        return nil;  // no query here
+    }
+    
+    if (!indexName) {
+        return nil;
+    }
+    
+    if (![clause isKindOfClass:[NSDictionary class]]) {
+        return nil; // should never get here as this would not pass normalization
+    }
+    
+    NSDictionary *textClause = (NSDictionary *) clause;
+    NSDictionary *searchClause = textClause[TEXT];
+    
+    NSString *tableName = [CDTQIndexManager tableNameForIndex:indexName];
+    NSString *search = [searchClause[SEARCH] stringByReplacingOccurrencesOfString:@"'"
+                                                                       withString:@"''"];
```

Comment:
The short answer here is that the escaping is not necessary in either iOS or Android.  The functionality is not affected by the escaping but...

Removed escaping in 09d8a4e530f4bd4a50e24d1ae14149b1e27dd27d.

Added FB case 47078 to remove from sync-android.